bundles/org.palladiosimulator.mdsdprofiles.api/src/org/palladiosimulator/mdsdprofiles/api/StereotypeAPI.java

```diff
@@ -367,6 +367,41 @@ public static StereotypeApplication getStereotypeApplication(final EObject stere
         return (DATA_TYPE) stereotypeApplication.eGet(taggedValue);
 
     }
+        
+    /**
+     * Returns the tagged value of the specified {@link EObject}. This ensures that both the {@link EMFProfile}
+     * and the {@link Stereotype} are applied to it. If this is not the case, an empty {@link Optional} is returned.
+     * 
+     * @param <DATA_TYPE> the data type of the tagged value
+     * @param stereotypedElement the entity on which the stereotype is applied
+     * @param taggedValueName the tagged value`s name
+     * @param stereotypeName the stereotype`s name
+     * @return the tagged value of the specified {@link EObject}
+     * @see #getTaggedValue(EObject, String, String)
+     */
+    @SuppressWarnings("unchecked")
+    public static <DATA_TYPE> Optional<DATA_TYPE> getTaggedValueSafe(final EObject stereotypedElement, final String taggedValueName, final String stereotypeName) {
+        if (stereotypedElement == null || taggedValueName == null || stereotypeName == null) {
+            return Optional.empty();
+        }
+
+        if (hasStereotypeApplications(stereotypedElement) == false || isStereotypeApplied(stereotypedElement, stereotypeName) == false) {
+            return Optional.empty();
+        }
+
+        /*
+         * This is definitely not the finest possibility, but unfortunately it is currently the only feasible one.
+         * Since several other methods of this StereotypeAPI signalize via RuntimeExceptions that EMFProfile
+         * and/or Stereotype are not applied.
+         */
```

Comment:
Indeed, it is not nice. But why do you need to catch a RuntimeException? As far as I understood, getStereotypeApplications does not throw a RuntimeException. If the remaining calls avoid NPEs, there should be no need to catch Exceptions.

Reply:
Yes, that's right, the methods `getStereotypeApplications` does not throw a `RuntimeException`. However, in the `StereotypeAPI` it is common practice to signal exceptions using `RuntimeException`s. For example, in the methods `unapplyStereotype`, `isStereotypeApplicable` and `getStereotypeApplication`. It would be a possibility to replace it with a specific exception of its own, but this should be a separate task instead of integrating this investigation in this issue. The block is intended as a safety precaution for future API changes. Will make this intention clear as a comment and resubmit my changes.